### PR TITLE
Update temperature reading docs

### PR DESCRIPTION
**Pull request by MSECode**
Update documentation for temperature reading done w/ sensors PT100 and PT1000, TDB and 2FOC.
Now documentation explaining how the reading of the temperature for ergoCub motors works is complete.

### CODE REVIEW

**Comment by pattacini** [comment]
Thanks @MSECode for your contribution.
I'll be reviewing it in the next few days.

**Comment by valegagge** [comment]
Hi @MSECode ,
before publishing the PR please recheck the documentation. I added some fixes and removed the error table because it was malformed and the errors were already mentioned in the `error handling` section. I just report the link in the index to make it more visible to the user.

**Review by valegagge — APPROVED** [review]
LGTM

File: docs/temperature_sensors/general/overview.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
+The Motor Temperature Sensors are simple sensors made of a specific type of resistor (PT100 or PT1000) and of a small board defined as the **Temperature Detection Board (TDB)**, which has the goal of decoding the value read by the resistor giving out a raw output code, which can be then translated to Celsius Degree so that it can be easily understood by the end-user.
+In a nutshell, most of the `ergoCub` motors are provided with an internal resistance temperature detector, which is then connected to a TDB board. This one is then linked to a `2FOC` board, which elaborates that data coming from the `I2C` and sends them via `CAN` to an `EMS` board. Finally, this one streams the raw temperature values via `ETH` to `embObjMotionControl` which updates and stores all of these data, ready to be requested by a user external module.
```
Comment (pattacini) [inline docs/temperature_sensors/general/overview.md:4]:
```suggestion
In a nutshell, most of the `ergoCub` motors are provided with an internal resistance temperature detector, which is then connected to a TDB board. The TDB board is then linked to a `2FOC` board, which elaborates the data acquired via `I2C` bus and sends it via the `CAN` bus to an `EMS` board. Finally, the EMS board streams the raw temperature values via the `ETH` bus to `embObjMotionControl`, which in turn updates and stores these acquisitions that the user can readily request from an external module.
```

File: docs/temperature_sensors/general/overview.md [hunk 2]
```diff
@@ -14,9 +14,11 @@ To correctly enable the use of the temperature sensors inside the configuration
 That being done, it should be possible to use the features of the temperature sensors.
 
 ### How to read the motor temperature
-The user needs to update the robot configuration files accordingly, as described in the [configuration section](#how-to-update-the-configuration-files), after ensuring that the sensor has been mounted on the desired motors.
-The correct configuration ensures that the motor temperature is read and forwarded to yarprobotinterface, but currently this measure in not published on the yarp `stateExt` port. Therefore the user needs to develop a module application similar to [this one](https://github.com/MSECode/motionControlModule), in order to read the temperature.
+First of all, the user should be sure that the sensor has been mounted on the desired motors, and then he/she needs to update the robot configuration files accordingly, as described in the [configuration section](#how-to-update-the-configuration-files).
```
Comment (pattacini) [inline docs/temperature_sensors/general/overview.md:17]:
```suggestion
First off, the user ought to make sure that the sensor is mounted on the desired motors, and then needs to update the robot configuration files accordingly, as described in the [configuration section](#how-to-update-the-configuration-files). 
```

File: docs/temperature_sensors/general/overview.md [hunk 3]
```diff
@@ -14,9 +14,11 @@ To correctly enable the use of the temperature sensors inside the configuration
...
-The user needs to update the robot configuration files accordingly, as described in the [configuration section](#how-to-update-the-configuration-files), after ensuring that the sensor has been mounted on the desired motors.
-The correct configuration ensures that the motor temperature is read and forwarded to yarprobotinterface, but currently this measure in not published on the yarp `stateExt` port. Therefore the user needs to develop a module application similar to [this one](https://github.com/MSECode/motionControlModule), in order to read the temperature.
+First of all, the user should be sure that the sensor has been mounted on the desired motors, and then he/she needs to update the robot configuration files accordingly, as described in the [configuration section](#how-to-update-the-configuration-files). 
+The correct configuration enables the motor temperature reading and sending the values to the yarprobotinterface, but, currently, these measures are not published on the yarp `stateExt` port. Therefore the user needs to develop a module application similar to [this one](https://github.com/MSECode/motionControlModule), to read the temperature.
```
Comment (pattacini) [inline docs/temperature_sensors/general/overview.md:18]:
```suggestion
The correct configuration allows reading and sending the motor temperature values to the `yarprobotinterface`. However, these measurements are not currently published on the YARP `stateExt` port. Therefore, the user needs to devise a module application, similar to [this one](https://github.com/MSECode/motionControlModule), to read the temperature.
```

File: docs/temperature_sensors/general/overview.md [hunk 4]
```diff
@@ -14,9 +14,11 @@ To correctly enable the use of the temperature sensors inside the configuration
...
 
 In addition, the user module can retrieve the `warningTemperatureLimits` and check if the temperature exceeds that limit to put the motor in a safer state with lower current consumption.
-Is it important to notice that the firmware puts the motor in fault when the temperature goes beyond the `hardwareTemperatureLimits`.
-More details on the temperature flow are available in the section [data flow](../software/dataflow.md).
+Is it important to notice that the firmware puts the motor in fault when the temperature goes beyond the `hardwareTemperatureLimits` for longer than 10 seconds.
```
Comment (pattacini) [inline docs/temperature_sensors/general/overview.md:21]:
```suggestion
!!! warning "HW Fault due to overheating"

    Is it important to notice that the firmware puts the motor in HW Fault when the temperature 
    goes beyond the `hardwareTemperatureLimits` for longer than 10 seconds.
```

File: docs/temperature_sensors/software/dataflow.md [hunk 5]
```diff
@@ -2,15 +2,10 @@
...
-The TBD receives the `Ohm` value from the `PT100` (or `PT1000` for different types of motors) and streams out with the raw data using the `I2C` protocol. This raw data is then read by the `2FOC` board with a frequency of 200Hz. The `2FOC` also checks if there is any `overHeating` error, relying on the user-defined configuration parameter `hardwareTemperatureLimits`. The raw temperatures are sent over `CAN` to the `EMS` board, which stores them in memory and sends to `embObjMotionControl` together with all the other motor information. 
-
+In a nutshell, as introduced in the [introductory section](../general/overview.md#temperature-sensors), most of the `ergoCub` motors are currently provided with a `PT100`, which is a particular resistor able to correlate a value of temperature to a value of resistance in `Ohm`. The resistor is connected to a board called **Temperature Detection Board (TDB)**.
```
Comment (pattacini) [inline docs/temperature_sensors/software/dataflow.md:5]:
```suggestion
In brief, as reported in the [introductory section](../general/overview.md#temperature-sensors), most of the `ergoCub` motors are currently provided with a `PT100`, which is a particular resistor able to correlate a value of temperature to a value of resistance in `Ohm`. The resistor is connected to a board called **Temperature Detection Board (TDB)**.
```

File: docs/temperature_sensors/software/dataflow.md [hunk 6]
```diff
@@ -2,15 +2,10 @@
...
+The TBD receives the `Ohm` value from the `PT100` (or `PT1000` for different types of motors) and streams out with the raw data using the `I2C` protocol. This raw data is then read by the `2FOC` board with a frequency of 10Hz. Then, as soon as the raw temperatures are read, they are sent over `CAN` bus to the `EMS` board (therefore the sending is done every `100ms` ). The `EMS` stores the received data in memory and sends to `embObjMotionControl` together with all the other motor information.
```
Comment (pattacini) [inline docs/temperature_sensors/software/dataflow.md:6]:
```suggestion
The TBD reads the `Ohm` value from the `PT100` (or `PT1000` for different types of motors) and streams out the raw data using the `I2C` protocol. This raw data is then received by the `2FOC` board at a frequency of `10 Hz` and, immediately after, handed over to the `EMS` board via the `CAN` bus. The `EMS` stores the temperature values in memory and sends them to `embObjMotionControl` together with all the other motor information.
```

File: docs/temperature_sensors/software/dataflow.md [hunk 7]
```diff
@@ -20,3 +15,23 @@ Here's a small yet exhaustive sequence diagram showing the whole dataflow as sum
     <img  src="../img/MotorTemperature-Feature.png" width="700">
 </p>
 
+
+### Error handling
+Mainly the error management is performed by the `2FOC`.
```
Comment (pattacini) [inline docs/temperature_sensors/software/dataflow.md:20]:
```suggestion
The error management is mainly performed by the `2FOC`.
```

File: docs/temperature_sensors/software/dataflow.md [hunk 8]
```diff
@@ -20,3 +15,23 @@ Here's a small yet exhaustive sequence diagram showing the whole dataflow as sum
...
+It performs some checks while reading the temperature from the `I2C` and specific negative values (defined in the code) are set to notify the user. The TDB sends 3 bytes on the I2C bus: the temperature value (2 bytes) and its configuration (1 byte). The `2FOC` performs the check mainly on the configuration byte.
```
Comment (pattacini) [inline docs/temperature_sensors/software/dataflow.md:21]:
```suggestion

The `2FOC` performs some checks while reading the temperature from the `I2C` and makes use of specific negative values (defined in the code) to notify the user of certain events.

The TDB board sends 3 bytes on the `I2C` bus: the temperature value (2 bytes) and its configuration (1 byte). The `2FOC` verifies mainly the configuration byte.
```

File: docs/temperature_sensors/software/dataflow.md [hunk 9]
```diff
@@ -20,3 +15,23 @@ Here's a small yet exhaustive sequence diagram showing the whole dataflow as sum
...
+
+
+The possible errors currently managed are the following:
+
+- `-90` : the `2FOC` cannot read from the `I2C`, meaning that the `ACK` is not received from the `TDB`
```
Comment (pattacini) [inline docs/temperature_sensors/software/dataflow.md:26]:
```suggestion
- `-90` : the `2FOC` cannot read from the `I2C`, meaning that the `ACK` is not received from the `TDB`.
```

File: docs/temperature_sensors/software/dataflow.md [hunk 10]
```diff
@@ -20,3 +15,23 @@ Here's a small yet exhaustive sequence diagram showing the whole dataflow as sum
...
+- `-70` :  the reading cannot be done for 10 consecutive seconds. An overheating fault is triggered.
```
Comment (pattacini) [inline docs/temperature_sensors/software/dataflow.md:27]:
```suggestion
- `-70` : the reading cannot be done for 10 consecutive seconds. An overheating fault is triggered.
```

File: docs/temperature_sensors/software/reading_data.md [hunk 11]
```diff
@@ -1,8 +1,8 @@
 ## Reading Raw Temperature Data
 
 Each temperature sensor outputs an integer raw value whose range depends on the type of resistance thermometer mounted on the motor.
-Currently we can configure 2 types of sensors: `PT100` and `PT1000`.
-For `PT100`, raw values typically range from `0` to `3500`. Instead, for `PT1000`, the range is between `0` and `8000`. At any rate, it should be noted that negative temperatures are accepted, even if it would be extremely rare to obtain them in normal working conditions. 
+Currently, we can configure 2 types of sensors: `PT100` and `PT1000`.
+For `PT100`, raw values typically range from `0` to `3500`. Instead, for `PT1000`, the range is between `0` and `8000`. At any rate, it should be noted that negative temperatures are accepted, but it is impossible to obtain them in normal working conditions. For this, we use them for notify error situations to the user.
```
Comment (pattacini) [inline docs/temperature_sensors/software/reading_data.md:5]:
```suggestion
For `PT100`, raw values typically range from `0` to `3500`. Instead, for `PT1000`, the range is between `0` and `8000`.

Although it is very unlikely to read negative temperatures in Celsius degrees during normal working conditions, we do accept negative values as they are specifically employed to notify the user of error conditions.
```

File: docs/temperature_sensors/software/reading_data.md [hunk 12]
```diff
@@ -11,7 +11,14 @@ Regarding the correlation between raw temperature data and Celsius degree values
 First of all, in order to reduce the effort of the end-user and at the same time remain compliant with the rest of the firmware, we have decided to work with values in Celsius degree on `embObjMotionControl` and, instead, use the raw value in the firmware part.
 However, the configuration files are always filled with values in Celsius degrees; likewise, the values printed on the YARP port are in the same format.
 By contrast, on the board level, we use raw values.
-All the applied conversions strictly depend on the physical components involved, including the voltage divider, the resistance thermometer as well as the ADC stage of the TDB board.
+All the applied conversions strictly depend on the physical components involved, including the voltage divider, the resistance thermometer as well as the ADC stage of the TDB board. 
+One important note to highlight is that even if we know that the conversion between raw and Celsius degree values is not exactly linear, we have decided to linearize the conversion by accepting an error of around ±1.5 ℃ over the 100 ℃.
```
Comment (pattacini) [inline docs/temperature_sensors/software/reading_data.md:15]:
```suggestion
One important note to highlight is that even if we know that the conversion between raw and Celsius degree values is not exactly linear, we have decided to linearize the conversion by accepting an error of around ±1.5 ℃ over 100 ℃. 
```

File: docs/temperature_sensors/software/reading_data.md [hunk 13]
```diff
@@ -11,7 +11,14 @@ Regarding the correlation between raw temperature data and Celsius degree values
...
 However, the configuration files are always filled with values in Celsius degrees; likewise, the values printed on the YARP port are in the same format.
 By contrast, on the board level, we use raw values.
-All the applied conversions strictly depend on the physical components involved, including the voltage divider, the resistance thermometer as well as the ADC stage of the TDB board.
+All the applied conversions strictly depend on the physical components involved, including the voltage divider, the resistance thermometer as well as the ADC stage of the TDB board. 
+One important note to highlight is that even if we know that the conversion between raw and Celsius degree values is not exactly linear, we have decided to linearize the conversion by accepting an error of around ±1.5 ℃ over the 100 ℃. 
+Anyways, if you are interested in the specific insight regarding the conversion you can check [this repository](https://github.com/MSECode/temperatureConversionTester), which has been mainly used for two scopes:
+
+- to check the validity of the linearization done and understand the magnitude of the error inside the working conditions, which are typically between 20 and 80 ℃
```
Comment (pattacini) [inline docs/temperature_sensors/software/reading_data.md:18]:
```suggestion
- to check the validity of the linearization done and understand the magnitude of the error inside the working conditions, which are typically between 20 and 80 ℃.
```

**Review by pattacini — APPROVED** [review]
Thanks.
I only applied some _cosmetic_ amendments to improve reading.

**Pull request merged**